static/js/subs.js

```diff
@@ -781,16 +779,21 @@ function hide_subscribe_toggle_spinner(row_object) {
     loading.destroy_indicator(spinner);
 }
 
-function ajaxSubscribe(stream, color, row_object) {
+function ajaxSubscribe(stream, color, is_web_public, row_object) {
     // Subscribe yourself to a single stream.
     let true_stream_name;
 
     if (row_object !== undefined) {
+        console.log(row_object);
         display_subscribe_toggle_spinner(row_object);
     }
+
+    const data = {};
+    data.subscriptions = JSON.stringify([{name: stream, color: color}]);
+    data.is_web_public = JSON.stringify(is_web_public);
```

Comment:
Should this flag be for each subscription?

Reply:
I think yes because I removed the `@require_non_guest_user` decorator of the `add_subscriptions_backend` function making it possible for a guest user to subscribe to a stream. So I'm using this flag to check if the user has permission to subscribe, I decided to pass this flag in the request so I don't have to query the db for the stream to check only if it is `web public`.

Reply:
> I decided to pass this flag in the request so I don't have to query the db for the stream to check only if it is web public.

I haven't read the actual code below yet, but a malicious actor could easily send an API request with `data.is_web_public=true` for any stream and get subscribed. We should not trust data from the frontend or api for things we can check from the database.

Reply:
Overall, I'm against this particular change. We should do all the sanity checks on the backend itself.

Reply:
Ok, so I've made a couple of manual tests for this. I tried to modify the HTTP request and pass the `is_web_pulbic = true` (as a normal user trying to subscribe to a private stream and as a guest user trying to subscribe to a public stream) and it didn't work. Then I hardcoded `is_web_public = true` on the `ajaxSubscribe` function and it didn't work as well. I also hardcoded `is_web_public = True` when it enters the `add_subscriptions_backend` function and it also didn't work. So, I guess that is no problem with this? @aero31aero @timabbott What do you think?

Reply:
![Captura de tela de 2020-05-13 19-26-20](https://user-images.githubusercontent.com/15564640/81873022-42925f80-9551-11ea-9413-49a163011d88.png)